### PR TITLE
Add table with search functionality to collections

### PR DESCRIPTION
**Pull request by jwallace94**
# Overview

# Related issue / discussion

# Additional information

# Dependency policy (RFC2)

- [ ] I have ensured that this PR meets [RFC2](https://pygeoapi.io/development/rfc/2) requirements

# Updates to public demo

- [ ] I have ensured that breaking changes to the [pygeoapi master demo server](https://github.com/geopython/demo.pygeoapi.io) have been addressed
  - [ ] https://github.com/geopython/demo.pygeoapi.io/blob/master/services/pygeoapi_master/local.config.yml

# Contributions and licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [ ] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution
- [ ] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

File: pygeoapi/templates/collections/items/index.html [hunk 1]
```diff
@@ -140,6 +146,32 @@ <h1>{% for l in data['links'] if l.rel == 'collection' %} {{ l['title'] }} {% en
...
+      	$('#searchTable thead th').each(function(i) {
+        	var title = $('#searchTable thead th').eq($(this).index()).text();
+        	$(this).html('<input type="text" placeholder="Search ' + title + '" data-index="' + i + '" />');
+      	});
+
+      	// DataTable
+      	var table = $('#searchTable').DataTable({
+        	responsive: true,
+        	paging: true,
+        	sorting: false,
```
Comment (shyni-s) [inline pygeoapi/templates/collections/items/index.html:161]: Will setting this to true enable sorting? If so, that's an easy win.

File: pygeoapi/templates/collections/items/index.html [hunk 2]
```diff
@@ -76,7 +82,7 @@ <h1>{% for l in data['links'] if l.rel == 'collection' %} {{ l['title'] }} {% en
 
       <div class="col-sm-12 col-md-6" style="overflow-x: scroll;">
         {% set props = [] %}
-        <table class="table table-striped table-bordered">
+        <table id="searchTable" class="stripe cell-border order-column" style="width:100%">
```
Comment (tWangchen) [inline pygeoapi/templates/collections/items/index.html:85]: Please ensure existing id naming convention is followed. Thank you.

File: pygeoapi/templates/collections/items/index.html [hunk 3]
```diff
@@ -140,6 +146,32 @@ <h1>{% for l in data['links'] if l.rel == 'collection' %} {{ l['title'] }} {% en
     </div>
     {% endif %}
     </section>
+    <script>
+    	$(document).ready(function() {
+      	// Setup - add a text input to each footer cell
+      	$('#searchTable thead th').each(function(i) {
```
Comment (tWangchen) [inline pygeoapi/templates/collections/items/index.html:152]: Please ensure existing id naming convention is followed. Thank you.

File: pygeoapi/templates/collections/items/index.html [hunk 4]
```diff
@@ -140,6 +146,32 @@ <h1>{% for l in data['links'] if l.rel == 'collection' %} {{ l['title'] }} {% en
...
+        	var title = $('#searchTable thead th').eq($(this).index()).text();
+        	$(this).html('<input type="text" placeholder="Search ' + title + '" data-index="' + i + '" />');
+      	});
+
+      	// DataTable
+      	var table = $('#searchTable').DataTable({
```
Comment (tWangchen) [inline pygeoapi/templates/collections/items/index.html:158]: Please ensure existing id naming convention is followed. Thank you.

File: pygeoapi/templates/collections/items/index.html (hunk 1 above)
Comment (jwallace94) [reply pygeoapi/templates/collections/items/index.html:161]: There's an issue where when you click in the search bar to type what you want to find it also sorts by that column which I am unsure how to address right now.

**Pull request merged**